### DISCUSSION TITLE
Modal height based on content inside

### DISCUSSION THREAD

**Discussion by AkshayAdiga1997** [software-mansion/react-native-screens#829] [Q&A] [answered]
I am using
`stackPresentation: 'modal'` configuration.
My modal content occupies around 30% of the top screen and is empty below. Is there way to make the modal adjust height based on the content inside ? So I would not want the modal to take 30% height. If possible, instead of me defining the percentage or value, can it do it automatically based on the content inside (Allowing keyboard push)?

**Comment by WoLewicki** [comment]
You can try `containedModal` and `transparentContainedModal` options and see if it is what you need.

**Reply by AkshayAdiga1997** [reply]
But then I won't get the effect of the behind screen getting zoomed out a bit (as seen in "modal" IOS). Also, is there a way to implement that modal in android? Thanks.

**Reply by WoLewicki** [reply, marked as answer]
I am not sure if iOS supports the presentation you are talking about in any different way than the normal one, although if you know any native application that uses it, please share it here. I am afraid Android doesn't have this concept of modals available. Both these issues could probably be implemented with applying custom transitions and layout of screens etc. but it is not available by default. If you don't want platform-specific implementations of navigators, the normal `stack` is the way to go for you since it uses only JS to implement its views and therefore is platform-agnostic.

**Reply by AkshayAdiga1997** [reply]
Okay thanks.
